api/graphql/schema.graphql

```diff
@@ -433,948 +442,1223 @@ enum GenericJobTitleKey {
+    where: PoolCandidateFilterInput
+    orderBy: [OrderByClause!] @orderBy
+    expiryStatus: CandidateExpiryFilter
+  ): [PoolCandidate]!
+    @all(scopes: ["expiryFilter", "available"])
+    @orderBy(column: "created_at", direction: DESC)
+    @guard
+    @can(ability: "viewAny")
+  classification(id: ID! @eq): Classification @find
+  classifications: [Classification]! @all
+  cmoAsset(id: ID! @eq): CmoAsset @find
+  cmoAssets: [CmoAsset]! @all
+  department(id: ID! @eq): Department @find
+  departments: [Department]! @all
+  poolCandidateFilter(id: ID! @eq): PoolCandidateFilter
+    @find
+    @guard
+    @can(ability: "view", query: true)
+  poolCandidateFilters: [PoolCandidateFilter]!
+    @all
+    @guard
+    @can(ability: "viewAny")
+  applicantFilter(id: ID! @eq): ApplicantFilter
+    @find
+    @guard
+    @can(ability: "view", query: true)
+  applicantFilters: [ApplicantFilter]! @all @guard @can(ability: "viewAny")
+  poolCandidateSearchRequest(id: ID! @eq): PoolCandidateSearchRequest
+    @find
+    @guard
+    @can(ability: "view", query: true)
+  poolCandidateSearchRequests: [PoolCandidateSearchRequest]!
+    @all
+    @guard
+    @can(ability: "viewAny")
+    @orderBy(column: "created_at", direction: DESC)
+  latestPoolCandidateSearchRequests(
+    limit: Int @limit
+  ): [PoolCandidateSearchRequest]!
+    @all
+    @orderBy(column: "created_at", direction: DESC)
+    @guard
+    @can(ability: "viewAny")
+  skillFamily(id: ID! @eq): SkillFamily @find
+  skillFamilies: [SkillFamily]! @all
+  skill(id: ID! @eq): Skill @find
+  skills: [Skill]! @all
+  genericJobTitle(id: ID! @eq): GenericJobTitle @find
+  genericJobTitles: [GenericJobTitle]! @all
 }
 input ClassificationBelongsTo {
-    connect: ID
-    disconnect: Boolean
+  connect: ID
+  disconnect: Boolean
 }
 input PoolsHasMany {
-    create: [CreatePoolInput]
+  create: [CreatePoolInput]
 }
 """
 When creating a User, name is required.
 """
 input CreateUserInput {
-    sub: String @rules(
-        apply: ["sometimes","unique:users,sub"]
-        messages: [{
-            rule: "unique"
-            message: "This user identifier (sub) is already in use"
-        }]
+  sub: String
+    @rules(
+      apply: ["sometimes", "unique:users,sub"]
+      messages: [
+        {
+          rule: "unique"
+          message: "This user identifier (sub) is already in use"
+        }
+      ]
     )
-    roles: [Role] = []
-
-    # Personal info
-    firstName: String! @rename(attribute: "first_name")
-    lastName: String! @rename(attribute: "last_name")
-    email: Email @rules(
-        apply: ["sometimes","unique:users,email"],
-        messages: [{
-            rule: "unique"
-            message: "This email address is already in use"
-        }]
+  roles: [Role] = []
+
+  # Personal info
+  firstName: String! @rename(attribute: "first_name")
+  lastName: String! @rename(attribute: "last_name")
+  email: Email
+    @rules(
+      apply: ["unique:users,email"]
```

Comment:
Did you mean to remove the sometimes?  It now prevents more than one user from having a blank email.  At least to the API, email is an optional field.
![image](https://user-images.githubusercontent.com/8978655/192867625-5c3fa2f0-9791-49f4-8bbe-e8b37798960c.png)

Reply:
Nope, I did not mean to remove that 😆 thanks for catching that. Re-added in [readd sometimes to unique email when creating user](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4103/commits/74711c3d02ce4fff92bb7afa7d2fadec1b5cd6ad)

Reply:
Shoot, this is still happening to me.  Is it fixed for you?

Reply:
Oh wait, I can create a user with no email now, but I can't update a user to have no email. :confused:

Reply:
Oh, I think I know what is going on. We need to prevent the email from being sent when it is empty. I assume this happens on all user forms.

Reply:
Okay, it should work now with [prevent email from being passed when empty](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4103/commits/81b8c5852ef095d0011239a6b1eb126f3ee3582e)

Reply:
Still no dice.  :cry: 
![image](https://user-images.githubusercontent.com/8978655/192875750-c863012e-0b2f-4ec1-a82b-2980a4c72a1a.png)
![image](https://user-images.githubusercontent.com/8978655/192875844-4b3845b9-aad4-4ad8-ad6c-0b1a83015b2a.png)

Reply:
Update it to be `null` instead of `undefined` and added `nullable` to email and it looks to be working 🤞 

[make email field null when empty](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4103/commits/5a3f66969eebd65a1e960b06d8bb3d5c4bc7a805)